### ISSUE TITLE
Better sorting the children of project node

### ISSUE THREAD

**Issue by jdneo** [microsoft/vscode-java-dependency#738]
LGTM overall. One small thing: can we adjust the order of source sets? Showing main/test source sets before the generated source sets would be better. 

after: 
![image](https://user-images.githubusercontent.com/45906942/230837135-3754c8e1-99ee-4c7e-99f3-411dc94e8851.png)

before:
![image](https://user-images.githubusercontent.com/45906942/230837179-7c882ca0-54bd-4929-82a0-5f5c2c2fe25d.png)

_Originally posted by @CsCherrYY in https://github.com/microsoft/vscode-java-dependency/pull/730#pullrequestreview-1377119435_

**Comment by jdneo** [comment]
One solution is sort at server side, which has the following order:

source[non-test] > source[test] > source[optional] > containers[JDK] > other containers

**Comment by Eskibear** [comment]
Is it possible to sort at client side, which would provide more flexibility? And the server side only provides type&context of the nodes.

**Comment by jdneo** [comment]
> Is it possible to sort at client side, which would provide more flexibility? And the server side only provides type&context of the nodes.

Yes, as long as the required information of the package roots can be passed to the client, then we can do it at the client side.
